### ISSUE TITLE
Use of autoreject for ECoG and SEEG data

### ISSUE THREAD

**Issue by AlexLepauvre** [autoreject/autoreject#198]
Hello everyone,

I am working on a project collecting a rather large amount of ECoG and SEEG data and would therefore be highly interested in using your method to detect and remove artifacts from the data. So far, we have been relying on computations of mean and standard error within channels, removing trials exceeding 4 times the standard deviation. The code is however always custom made and doesn't seem sensitive enough.

I would like to know first of all is if from a theoretical perspective the method makes sense at all to use, at least the threshold detection part? My understanding is that the local threshold detection method would be the way to go, as in intracranial recordings, amplitudes are widely varied across electrodes location. And even more so given the idiosyncratic electrodes coverage in most cases. Therefore, detecting one threshold per location would be best, and shouldn't be a problem with your method, right?

The next question is about what happens to the trials with artifacts? Indeed, your method when using local thresholds uses rho and K parameters to decide whether channels should be interpolated, left as such or the trials to be discarded. In the case of intracranial recordings, the uneven coverage makes interpolation very questionable. Different labs have different methods to deal with this, but I believe the most straight forward is to simply discard any trials in which one or more electrode has been detected as bad. While it has the downside on being very harsh, the overall low amount of noise in intracranial recordings might enable us to get away with it. Do you think simply setting the k to 1 is a viable option for our use case?

I have been playing around with the autoreject functions and noticed that if the channels type in mne are set to seeg or ecog, the functions will not work. A workaround would be to set the channels to eeg only for the autoreject and reset them to seeg and ecog after the fact, though it doesn't seem very elegant.

Thanks in advance for the support,

Kind regards,

Alex

**Comment by jasmainak** [comment]
I have this on my agenda too, great minds think alike ;-) I think this would be great to have. I have some private code and we can exchange ideas.

Sorry cannot give a detailed response right now due to a deadline, but how about we touch base next week? Feel free to ping me here and/or we can email or video chat.

**Comment by jasmainak** [comment]
> Therefore, detecting one threshold per location would be best, and shouldn't be a problem with your method, right?

correct, but you do need a mix of good and bad epochs. From my limited experience with sEEG, this probably shouldn't be a problem

> the overall low amount of noise in intracranial recordings

questionable claim and depends how you define noise :)

> Do you think simply setting the k to 1 is a viable option for our use case?

note k should be in fraction I believe. I think for intracranial you need to set some kind of a threshold for the number of trials. If there are less than x trials left, then discard the entire channel as bad. I'm not sure ... need to look at more intracranial data but I wouldn't have too much confidence in your results if you were to average only a few trials

> I have been playing around with the autoreject functions and noticed that if the channels type in mne are set to seeg or ecog, the functions will not work. A workaround would be to set the channels to eeg only for the autoreject and reset them to seeg and ecog after the fact, though it doesn't seem very elegant.

I think this should certainly be fixed! pull request is welcome :)

**Comment by chapochn** [comment]
Hello! I would also be interested in making it work with ECoG, and was wondering if you made any progress in this direction?

**Comment by jasmainak** [comment]
@chapochn sorry for the delayed response. Is there a public dataset that could be used to try this on? Maybe something on openneuro?

**Comment by chapochn** [comment]
There are many datasets in openneuro, but I woundn't know a specific one on which there would be artifacts to check on...
